### PR TITLE
Recreate default for use_summaries

### LINKED ISSUE #113: Summaries are being enabled when we aren't using a resolver
The `use_resolver = False` config should prevent us from ever creating a `SummaryResolver`

### PR DESCRIPTION
**Pull request by natebosch**
Fixes #113

We set a default value for use_summaries in macro, but the real value
for use_resolver may not have been available there if it was, for
instance, a `select`

We still want to attempt the defaulting in the macro when it is possible
since it also controls whether the SDK lib files are set as inputs to
the binary.

### CODE REVIEW

File: dart/build_rules/codegen/_codegen_binary.bzl [hunk 1]
```diff
@@ -27,10 +27,16 @@ def _codegen_binary_impl(ctx):
       ctx.attr.deps + [ctx.attr._codegen_dep],
       data = ctx.files.data,
   )
+
+  use_summaries = ctx.attr.use_summaries
+  # Redo default in case the macro did not have a real value for use_resolver
+  if not ctx.attr.use_resolver:
```
Comment (jakemac53) [inline dart/build_rules/codegen/_codegen_binary.bzl]: What if somebody explicitly passes use_summaries?

Comment (natebosch) [inline dart/build_rules/codegen/_codegen_binary.bzl]: It doesn't make sense to pass use_summaries = True and use_resolver = False. IMO I think it's ok for behavior to be wrong in that case.

Comment (jakemac53) [inline dart/build_rules/codegen/_codegen_binary.bzl]: It doesn't make sense to us, but I don't think many people understand things enough to know that?

**Comment by jakemac53** [comment]
talked offline and we will merge these attributes later into a single one

**Comment by natebosch** [comment]
Filed https://github.com/dart-lang/rules_dart/issues/115 to track future improvements to the API

**Pull request merged**